src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -669,6 +669,44 @@ void QgsWmsCapabilities::parseCapability( QDomElement const &e, QgsWmsCapability
     }
   }
 
+  // Layers in a WMS-C Capabilities TileSet do not have layer title or abstract
+  // still those could be present in a Layer list.
+  if ( !mTileLayersSupported.isEmpty() )
+  {
+    QHash<QString, QString> titles;
+    QHash<QString, QString> abstracts;
+
+    for ( int i = 0; i < mLayersSupported.size(); ++i )
+    {
+      if ( !mLayersSupported.at( i ).title.isEmpty() && !mLayersSupported.at( i ).name.isEmpty() )
+      {
+        titles.insert( mLayersSupported.at( i ).name, mLayersSupported[i].title );
+      }
+    }
+    for ( int i = 0; i < mTileLayersSupported.size(); ++i )
```

Comment:
In this case, you can use range-based loop too, but it's not a `const` loop this time.

Reply:
Done.